pydash/strings.py

```diff
@@ -736,6 +736,56 @@ def lines(text):
     return text.splitlines()
 
 
+def lower_case(text):
+    """Converts string to lower case as space separated words. Before
+    converting the given text to lower case this method:
+
+      - replaces special character with white spaces,
+      - adds a white space before the capital letter,
+      - replaces multiple whitespaces with a single whitespace,
+      - pertains spaces between words in the given text,
+      - pertains the digits given in the text.
+
+    Args:
+        text (str): String to convert.
+
+    Returns:
+        str: String converted to lower case as space separated words.
+
+    Example:
+
+        >>> lower_case('fooBar')
+        'foo bar'
+        >>> lower_case('--foo-Bar--')
+        'foo bar'
+        >>> lower_case('/?*Foo10/;"B*Ar')
+        'foo10 b ar'
+    """
+    # TODO: Add release version.
+    characters = []
+
+    for char in text:
+        if not char.isalpha() and not char.isdigit():
+            # If any of the characters is neither an alphabet or a digit, then
+            # replace them with whitespaces.
+            characters.append(' ')
+        elif char.istitle():
+            # If any of the characters is a capital letter, then go ahead and
+            # add a whitespace before the capital letter.
+            characters.append(' {0}'.format(char))
+        else:
+            characters.append(char)
+
+    text = ''.join(characters)
+    text = text.strip()
+
+    # Replace multiple whitespaces between the words in the text with a single
+    # whitespace.
+    text = re.sub("\s\s+", " ", text)
+
+    return text.lower()
```

Comment:
Which edge case doesn't `words()` handle?

```python
>>> from pydash import words
>>> def lower_case(text):
...     return ' '.join(words(text)).lower()
... 
>>> tests = (('fooBar', 'foo bar'),
... ('--foo-Bar--', 'foo bar'),
... ('*Foo*B_a*R', 'foo b a r'),
... ('*Foo10_B*Ar', 'foo 10 b ar'),
... ('/?*Foo10/;"B*Ar', 'foo 10 b ar'),
... ('/?F@O__o10456?>.B?>";Ar', 'f o o 10456 b ar'),
... ('FoO Bar', 'fo o bar'),
... ('F\n\soO Bar', 'f so o bar'),
... ('Foo1054665Bar', 'foo 1054665 bar'))
>>> for case, expected in tests:
...    lower_case(case) == expected
... 
True
True
True
True
True
True
True
True
True
```

The purpose of `words()` is to do all the things you mention. It may be a complicated regex, but it's what Lodash uses (NOTE: `words()` may need to be updated if Lodash has changed their regex since pydash was last updated).

Basically, any function in pydash that needs to break things up on "words" should use the `words()` function. All of the `*_case()` functions are using `words()` under the hood.